File: spl-net/src/main/java/bgu/spl171/net/impl/TFTP/CommandBuilders/DeleteRequestCommandBuilder.java
package bgu.spl171.net.impl.TFTP.CommandBuilders;

import bgu.spl171.net.impl.TFTP.Commands.Command;
import bgu.spl171.net.impl.TFTP.Commands.DeleteRequest;
import bgu.spl171.net.impl.TFTP.EncoderDecoders.StringEncoderDecoder;

public class DeleteRequestCommandBuilder implements CommandBuilder {
    StringEncoderDecoder stringEncoderDecoder = new StringEncoderDecoder();

    @Override
    public Command decodeNextByte(byte nextByte) {
        String fileName = stringEncoderDecoder.decodeNextByte(nextByte);
        if(fileName!=null){
            return new DeleteRequest(fileName);
        }
        return null;
    }
}

File: java to c++ builders/DeleteRequestCommandBuilder.cpp
#include "DeleteRequestCommandBuilder.h"

namespace bgu
{
	namespace spl171
	{
		namespace net
		{
			namespace impl
			{
				namespace TFTP
				{
					namespace CommandBuilders
					{
						using bgu::spl171::net::impl::TFTP::Commands::Command;
						using bgu::spl171::net::impl::TFTP::Commands::DeleteRequest;
						using bgu::spl171::net::impl::TFTP::EncoderDecoders::StringEncoderDecoder;

						Command *DeleteRequestCommandBuilder::decodeNextByte(char nextByte)
						{
							std::wstring fileName = stringEncoderDecoder->decodeNextByte(nextByte);
							if (fileName != L"")
							{
								return new DeleteRequest(fileName);
							}
							return nullptr;
						}
					}
				}
			}
		}
	}
}
